pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -327,9 +327,13 @@ stdenv.mkDerivation {
     ++ lib.optional (cc.langFortran or false) ./fortran-hook.sh;
 
   postFixup =
+    ''
+      export isClang=${if isClang then "true" else "false"}
```

Comment:
Please use `env = { inherit isClang; };` instead.

Reply:
Thank you for the suggestion. I will try it, though it will take me awhile to build things to test :)

Reply:
I still need to figure out fast ways to work on these sorts of packages, it's a fun challenge.

Reply:
I updated it to be:
```
env.isClang = if isClang then "true" else "false";
```
I want it to be a string literal since it's used as the "true" and "false" shell commands in the shell script:
```
if @isClang@; then
  ...
else
  ...
fi
```

Reply:
Whoa, that's cool. I've changed it to use that now. (When `isClang` is false, it ends up yielding `((   ))` but that still evaluates to false.) Thanks!

Reply:
Yeah, it's the Bash arithmetic syntax, very useful for flags/conditionals :)

That said, maybe it'd be best to just pass `-fwrapv` unconditionally? I think it makes strictly more programs defined on GCC and I doubt it has a meaningful performance penalty vs. `-fno-strict-overflow`. (But I'm not an expert here, so I could be wrong.)

Reply:
According to https://www.airs.com/blog/archives/120 the behaviour should only differ on (irrelevant to us) non-wrapping architectures, and `-fno-strict-overflow` may optimize better without changing behaviour. However, that was in 2008; the example yields identical object code with GCC 13. I struggle to think of what a substantive difference would be on the architectures we support. Perhaps `-fno-strict-overflow` is better as it conveys our hardening intent of "don't blow up the world if it happens, but overflow is still UB and a program bug". I hope someone who is more knowledgeable about GCC can chime in.

Reply:
@emilazy Do you mind if we merge this clang-only approach? I think your reasoning makes sense, but I don't know enough about what guides our choice for the default hardening flags to push for a conversation.

Reply:
Yeah the GCC thing seems like a rabbit hole :)